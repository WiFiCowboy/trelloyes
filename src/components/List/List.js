import React from 'react'
import Card from '../Card/Card'

function List(props) {
  console.log('List:',props)
  let cards = props.cards.map(card =>
    <Card key={props.id} title={card.title} content={card.content} />)

  return (
    <section className="List">
        <header className="List-cards">
            <h2>{props.header}</h2>
          </header>
        <div className="List-cards">{cards}</div>
    </section>
  )
}

export default List
