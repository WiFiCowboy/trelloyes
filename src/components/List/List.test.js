import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

describe('<List />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('matches snapshot', () => {
    const divSnap = renderer.create(<List />).toJSON();
    expect(divSnap).toMatchSnapshot();
  });
});