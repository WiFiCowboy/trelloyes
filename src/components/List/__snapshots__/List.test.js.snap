// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> matches snapshot 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      000
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
